cocos/spine/skeleton.ts

```diff
@@ -966,11 +966,10 @@ export class Skeleton extends UIRenderer {
                 this._animCache.updateToFrame(0);
                 this._curFrame = this._animCache.frames[0];
             }
-        } else {
-            this._animationName = name;
-            trackEntry = this._instance!.setAnimation(trackIndex, name, loop);
         }
-        this._markForUpdateRenderData();
+
+        this._animationName = name;
+        const trackEntry = this._instance!.setAnimation(trackIndex, name, loop);
```

Comment:
this._instance is not working at cache mode. 
This issue is scheduled to be resolved in an upcoming major version refactoring.